pkg/apis/core/validation/validation.go

```diff
@@ -5762,3 +5763,67 @@ func ValidateProcMountType(fldPath *field.Path, procMountType core.ProcMountType
 		return field.NotSupported(fldPath, procMountType, []string{string(core.DefaultProcMount), string(core.UnmaskedProcMount)})
 	}
 }
+
+var (
+	supportedScheduleActions = sets.NewString(string(core.DoNotSchedule), string(core.ScheduleAnyway))
+)
+
+// validateTopologySpreadConstraints validates given TopologySpreadConstraints.
+func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
```

Comment:
This is not same as original PR. Need to find out why

Reply:
This is a combine of PR 80395 (Optimize logic in EvenPodsSpread API validation) and 77327. It should be ok.